pkg/opentelemetry/otelhttpclient/client.go

```diff
@@ -0,0 +1,23 @@
+package otelhttpclient
+
+import (
+	"net/http"
+)
+
+func New(name string, client *http.Client) *http.Client {
+	if client == nil {
+		return NewClient(name)
+	}
+	return NewFromClient(client, name)
+}
+
+func NewClient(name string) *http.Client {
+	return &http.Client{
+		Transport: NewHTTPTransport(nil, name),
+	}
+}
+
+func NewFromClient(httpClient *http.Client, name string) *http.Client {
```

Comment:
this looks like it has the same signature with the `New` function. let's either move this logic to `New` or make this as unexported function